pandas/core/arrays/string_arrow.py

```diff
@@ -227,10 +227,22 @@ def _chk_pyarrow_available(cls) -> None:
 
     @classmethod
     def _from_sequence(cls, scalars, dtype: Dtype | None = None, copy: bool = False):
+        from pandas.core.arrays.masked import BaseMaskedArray
+
         cls._chk_pyarrow_available()
-        # convert non-na-likes to str, and nan-likes to ArrowStringDtype.na_value
-        scalars = lib.ensure_string_array(scalars, copy=False)
-        return cls(pa.array(scalars, type=pa.string(), from_pandas=True))
+
+        if isinstance(scalars, BaseMaskedArray):
+            # avoid costly conversion to object dtype in ensure_string_array and
+            # numerical issues with Float32Dtype
+            na_values = scalars._mask
+            result = scalars._data
+            result = lib.ensure_string_array(result, copy=False, convert_na_value=False)
```

Comment:
umm this doesn't copy? (its fine but then you are mutating)

Reply:
that's not a change in this PR. will need to look back why we do this.

Reply:
updated to be consistent with StringArray. I guess that the assumption was that a copy would probably have been created in most cases (unless an object array was passed) and explicitly avoid the additional copy made in ensure_string_array since that does not track if a copy/new array has been already been created before making a copy. The default for copy is False.